### PR TITLE
OPI builder bugfixes

### PR DESCRIPTION
**Pull request by jbobnar**
The pull request resolves the issues listed in #1682.

### CODE REVIEW

**Comment by berryma4** [comment]
@willrogers @shroffk 
I need another reviewer.

**Comment by willrogers** [comment]
Sorry, it's tricky to review because it's a wide range of things.

One thing I would say - I expect that if you build against Mars (not Mars M2) then perspective saving probably won't work correctly now that the versions have been lowered.  It's a bit difficult to test exactly what will happen.

If you build against Mars M2 you should get the latest version of plugins anyway so everything should work ok.

File: applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/META-INF/MANIFEST.MF [hunk 1]
```diff
@@ -35,7 +35,7 @@ Require-Bundle: org.eclipse.ui;resolution:=optional,
  org.python.jython;bundle-version="2.5.2",
  org.csstudio.swt.widgets;bundle-version="2.1.0",
  org.csstudio.swt.xygraph;bundle-version="2.0.4",
- org.eclipse.e4.ui.workbench;bundle-version="1.3.1"
+ org.eclipse.e4.ui.workbench;bundle-version="1.3.0"
```
Comment (berryma4) [inline applications/opibuilder/opibuilder-plugins/org.csstudio.opibuilder/META-INF/MANIFEST.MF]: here

File: core/ui/ui-plugins/org.csstudio.perspectives/META-INF/MANIFEST.MF [hunk 2]
```diff
@@ -6,19 +6,18 @@ Bundle-Version: 0.0.1.qualifier
 Eclipse-ExtensibleAPI: true
 Require-Bundle: org.eclipse.osgi.services;bundle-version="3.5.0",
  org.eclipse.e4.core.services;bundle-version="2.0.0",
- org.eclipse.e4.ui.workbench;bundle-version="1.3.1",
+ org.eclipse.e4.ui.workbench;bundle-version="1.3.0",
```
Comment (berryma4) [inline core/ui/ui-plugins/org.csstudio.perspectives/META-INF/MANIFEST.MF]: and here

**Comment by berryma4** [comment]
@willrogers are these the problem files for the perspective saving?
We need the perspective saving ....

**Comment by jbobnar** [comment]
Tycho will always take the latest from the p2 repository, so you should be safe. The plugin is compatible with both versions.

1.3.1 has one bugfix with respect to 1.3.0. I don't know if this bugfix is relevant or not. There might be additional bugs fixed in future releases, but probably nobody will go back to this plugin and raise this number to 1.4.0. You will just build the CSS using the newest version.

Anyway, I'll ask ITER if they can upgrade eclipse version.

**Comment by willrogers** [comment]
Yes, I think exactly that bugfix, if you look at this: https://git.eclipse.org/r/#/c/63125/

As Jaka says, I think we are safe.  I don't think ITER will get the perspective saving working correctly.  Jaka, how are you building against the older version?  The `core` and `applications` pom files pick up the newer versions.

**Comment by jbobnar** [comment]
Their build server does not have access to internet at all. All eclipse and other plugins were are packed into a p2 repository in advance. They control which versions of plugins are packed and right now its is Mars 0. The generated p2 is installed locally and that is the one that jenkins sees. In addition they also have their own artifactory server, which is a link between the build server and central maven repository, so that they can build things like maven-osgi-bundles.

**Pull request merged**